helm/generic-node/templates/generic-node.yaml

```diff
@@ -0,0 +1,283 @@
+---
+{{if .Values.daemon.enabled}}
+apiVersion: apps/v1
+{{- if .Values.features.persistence.enabled}}
+kind: StatefulSet
+{{- else}}
+kind: Deployment
+{{- end}}
+metadata:
+  name: {{.Values.name}}
+  labels:
+    app: {{.Values.name}}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: {{.Values.name}}
+  serviceName: "nginx"
+  template:
+    metadata:
+      labels:
+        app: {{.Values.name}}
+        testnet: {{.Values.testnetName}}
+        role: generic-node
+        class: "generic-node"
+        version: {{.Values.image.tag | quote}}
+        syncStatus: INIT
+      annotations:
+        prometheus.io/scrape: 'true'
+        prometheus.io/port: {{.Values.ports.metrics | quote}}
+        prometheus.io/path: '/metrics'
+    spec:
+      affinity:
+        nodeAffinity:
+          requiredDuringSchedulingIgnoredDuringExecution:
+            nodeSelectorTerms:
+              - matchExpressions:
+                - key: "cloud.google.com/gke-preemptible"
+                  operator: In
+                  values: ["true"]
+      initContainers:
+      {{- if .Values.libp2p.staticKeypair.enabled}}
+      - name: libp2p-init-my-service
+        image: busybox:latest
+        command: ['sh', '-c', '/bin/cp /libp2p-keys/* /root/libp2p-keys && /bin/chmod -R 0700 /root/libp2p-keys/']
+        volumeMounts:
+        - mountPath: /libp2p-keys
+          name: source-libp2p-keys
+        - mountPath: /root/libp2p-keys
+          name: destination-libp2p
+      {{- end }}
+      {{- if .Values.features.privateKeySecret.enabled}}
+      - name: fix-perms
+        image: busybox:latest
+        command: [ 'sh', '-c', 'for dir in keys echo-keys faucet-keys; do [ -d /$dir ] && /bin/cp /$dir/* /wallet-keys; done; /bin/chmod 0700 /wallet-keys']
+        volumeMounts:
+        - mountPath: "/keys/"
+          name: private-keys
+          readOnly: true
+        - mountPath: /wallet-keys
+          name: wallet-keys
+      {{- end }}
+      containers:
+      - name: daemon
+        image: "{{.Values.image.repository }}:{{.Values.image.tag}}"
+        imagePullPolicy: {{.Values.image.pullPolicy}}
+        {{- if .Values.features.customEntrypoint.enabled}}
+        command: [ {{ .Values.features.customEntrypoint.command}} ]
+        {{- end }}
+        volumeMounts:
+        - name: destination-libp2p
+          mountPath: /root/libp2p-keys
+          readOnly: true
+        - name: config-dir
```

Comment:
this config-dir mount should only be enabled if .Values.persistence.enabled is false